### PR TITLE
Initial work on i18n mechanics

### LINKED ISSUE #110: Add support for multiple languages (l18n)
This requested project improvement is to add support for multiple languages (l18n).  When people can interact with a VC in a language they easily understand it will help erase barriers to adoption. To make this happen I've implemented a the basic mechanics for internationalization (i8n) using two basic npm packages. One to help select the prefers language and another to manage translations. The implementation is not complete as all strings are not moved to the translations file but is good enough to start the conversation.

When done, I would expect the application to determine the current language and, if a matching translation is available, use it throughout the application. The base implementation will remain `en` and others will be added pending community contributions.

### PR DESCRIPTION
**Pull request by jleach**
# Summary of Changes

Implement a basic i18n strategy using `react-i18next` as per the related issue.

# Related Issues

Fixes #110

# Pull Request Checklist

This is just a reminder about the most common mistakes. Please make sure that you tick all _appropriate_ boxes. But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this).
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components.
- [x] Run prettier: `npm run style-format`
- [x] Updated **documentation** for changed code and new or modified features.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Review by TimoGlastra — COMMENTED** [review]
Is there a particular reason you went with localized-strings?

It hasn't been updated in 2 years and isn't one of the major libraries out there (AFAIK).

We use i18next and react-i18next in our wallet, maybe something to look into? Also fine with using this library, just curious for the reasoning

**Comment by jleach** [comment]
@TimoGlastra No reason other that it was a fairly simple implementation. I'll look into ` react-i18next` and see how it can be adapted to this project.

**Comment by manghuli** [comment]
would this take the language their phone is set to automatically?

**Comment by jleach** [comment]
@NeilSMyers That's the idea. It will find the best available matching translation based on what the phone is set for.

**Comment by manghuli** [comment]
@jleach what do you think of having a way to manually change the language? Idk if this use case is too small to worry about but if someone has their phone on English but wants the app in French for clarity's sake

**Comment by jleach** [comment]
@NeilSMyers I think its a great feature. Well placed in the "settings" once we get to implementing more functionality there. I've made a note about this and will check with the team if we're going to keep it in our board or use Issues (probably issues).

@TimoGlastra Switched to use `react-i18next`.

**Comment by TimoGlastra** [comment]
Awesome! 


I'll try to share some typescript goodies later today that can make the 't()' method fully typed. This way it will error if you put in a string that doesn't have a translation

File: App/i18n/en/index.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const translation: Record<string, string> = {
```
Comment (TimoGlastra) [inline App/i18n/en/index.ts]:
```suggestion
const translation = {
```

File: App/i18n/index.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { initReactI18next } from 'react-i18next'
+import * as RNLocalize from 'react-native-localize'
+import { defaultLanguage } from '../constants'
+import en from './en'
+
+const resources = {
+  en: {
+    translation: en,
+  },
+}
```
Comment (TimoGlastra) [inline App/i18n/index.ts:11]: Add a type `Translation` that is based on the `en` translation. Other languages can now set this as the type, meaning that if `en` adds a property we will get an error if it is not added for other languages.


```suggestion
export type Translation = typeof en
```

Then create a `types/react-i18next.d.ts` file and add the following to it:

```
import { Translation } from '../src/translations' # update with your path

import 'react-i18next'

declare module 'react-i18next' {
  interface CustomTypeOptions {
    resources: {
      translation: Translation
    }
  }
}
```

The `t()` method from `useTranslation()` is now fully typed

**Comment by TimoGlastra** [comment]
For adding typescript integration with react-i18next: Add a type `Translation` that is based on the `en` translation. Other languages can now set this as the type, meaning that if `en` adds a property we will get an error if it is not added for other languages.


```
export type Translation = typeof en
```

Other language:

```
const translation: Translation = {

}
```


Then create a `types/react-i18next.d.ts` file and add the following to it:

```
import { Translation } from '../src/translations' # update with your path

import 'react-i18next'

declare module 'react-i18next' {
  interface CustomTypeOptions {
    resources: {
      translation: Translation
    }
  }
}
```

The `t()` method from `useTranslation()` is now fully typed

**Comment by jleach** [comment]
@TimoGlastra Updated to include type support.

**Review by TimoGlastra — APPROVED** [review]
Nice!

**Pull request merged**